kratos/sources/model_part_io.cpp

```diff
@@ -25,4743 +25,5056 @@
 
-    ///@}
-    ///@name Operations
-    ///@{
+ModelPartIO::~ModelPartIO() {
+    if (mOptions.IsNot(IO::SKIP_TIMER)) Timer::CloseOuputFile();
+}
 
-    bool ModelPartIO::ReadNode(NodeType& rThisNode)
-    {
-        KRATOS_ERROR << "Calling base class member. Please check the definition of derived class." << std::endl;
-    }
+/***********************************************************************************/
+/***********************************************************************************/
 
-    bool ModelPartIO::ReadNodes(NodesContainerType& rThisNodes)
-    {
-        KRATOS_TRY
-        ResetInput();
-        std::string word;
-        while(true)
-        {
-            ReadWord(word);
-            if(mpStream->eof())
-                break;
-            ReadBlockName(word);
-            if(word == "Nodes")
-                ReadNodesBlock(rThisNodes);
-            else
-                SkipBlock(word);
-        }
+bool ModelPartIO::ReadNode(NodeType& rThisNode)
+{
+    KRATOS_ERROR << "Calling base class member. Please check the definition of derived class." << std::endl;
+}
 
-        return true;
-        KRATOS_CATCH("")
-    }
+/***********************************************************************************/
+/***********************************************************************************/
 
-    std::size_t ModelPartIO::ReadNodesNumber()
+bool ModelPartIO::ReadNodes(NodesContainerType& rThisNodes)
+{
+    KRATOS_TRY
+    ResetInput();
+    std::string word;
+    while(true)
     {
-        KRATOS_TRY;
-        ResetInput();
-        std::string word;
-        std::size_t num_nodes = 0;
-        while(true)
-        {
-            ReadWord(word);
-            if (mpStream->eof())
-                break;
-            ReadBlockName(word);
-            if(word == "Nodes")
-                num_nodes += CountNodesInBlock();
-            else
-                SkipBlock(word);
-        }
-
-        return num_nodes;
-        KRATOS_CATCH("");
+        ReadWord(word);
+        if(mpStream->eof())
+            break;
+        ReadBlockName(word);
+        if(word == "Nodes")
+            ReadNodesBlock(rThisNodes);
+        else
+            SkipBlock(word);
     }
 
-    void ModelPartIO::WriteNodes(NodesContainerType const& rThisNodes)
-    {
-        (*mpStream) << "Begin Nodes" << std::endl;
-        for(NodesContainerType::const_iterator it_node = rThisNodes.begin() ; it_node != rThisNodes.end() ; it_node++)
-            (*mpStream) << "\t" << it_node->Id() << "\t" << it_node->X()  << "\t" << it_node->Y() << "\t" << it_node->Z() << std::endl;
-        (*mpStream) << "End Nodes" << std::endl << std::endl;
-    }
+    return true;
+    KRATOS_CATCH("")
+}
 
-    void ModelPartIO::ReadProperties(Properties& rThisProperties)
+/***********************************************************************************/
+/***********************************************************************************/
```

Comment:
I think those are unnecessary, increase the file size for no reason...

Reply:
Makes easier to identify when a function starts and finishes. I see it very practical

Reply:
I agree with @philbucher that this separators are really unnecessary and only make the file less readable as there are less code per page and more scrolls

Reply:
Removed